### ISSUE TITLE
Infinite loop if busy message returned

### ISSUE THREAD

**Issue by eharris2000** [qmegas/memcache-search#1]
It looks like MemcacheSearch.php will get caught in an infinite loop if this message is returned from Memcached:

BUSY currently processing crawler request

I was able to work around this by:
Line 27: $busy = false;
Line 45: 
      elseif( $line === 'BUSY currently processing crawler request' ){
                       $busy = true;
                       break 2;

                    }

Line 63: if( $busy ){ throw new Exception\BusyException("Memcached server was busy."); }

I then copied ConnectionException to BusyException, and changed Class Name to match.

This allowed the loop to exit, close the socket, and return exception to caller.  Not sure if this is long term solution as you may just get this message back from one server not another but at least it will not get in infinite loop and gives caller a chance to sleep and try again.

**Comment by qmegas** [comment]
@eharris2000 Thanks for bug report. I'll check the problem and apply the fix.

**Comment by qmegas** [comment]
[Released version 0.1.3](https://github.com/qmegas/memcache-search/releases/tag/0.1.3)

You may now choose different strategies on server busy.

1. Ignore busy server
```php
$search->onBusyStrategy(\Qmegas\MemcacheSearch::ON_BUSY_IGNORE);
```

2. Throw exception (default strategy)
```php
$search->onBusyStrategy(\Qmegas\MemcacheSearch::ON_BUSY_EXCEPTION);
```

3. Wait for some time
```php
//Will wait 10 seconds for crawler to be available, if still not available then throw exception
$search->onBusyStrategy(\Qmegas\MemcacheSearch::ON_BUSY_TIMEOUT, 10);
```
